### PR TITLE
chore/add api helper for tests

### PR DESCRIPTION
**Pull request by dalebennett1992**
Add an API helper that can speed up our tests. Not currently used but the methods work. Will use this when @dexturr onboarding remembering where I am PR is merged

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
     "@types/chrome": "^0.0.231",
-    "@types/jest": "^29.5.0",
+    "@types/jest": "^29.5.1",
```
Comment (dexturr) [inline package.json:64]: chore (blocking): why no lock changes related to a package change? Can we run yarn and double check this is correct.

File: test/e2e/wallet-helpers/api-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  private driver: WebDriver
+
+  constructor(driver: WebDriver) {
+    this.driver = driver
+  }
+
+  async generateRecoveryPhrase() {
+    return await this.driver.executeScript<string>(async () => {
+      // @ts-ignore
+      const { recoveryPhrase } = await window.client.request('admin.generate_recovery_phrase', null)
```
Comment (dexturr) [inline test/e2e/wallet-helpers/api-helpers.ts:12]: chore (non-blocking): suggest adding a index.d.ts file to tell typescript that the window object has this client prop rather than requiring an ignore everywhere.

```
interface Window {
  client?: JSONRPCClient;
}
```

File: jest.config-e2e.js [hunk 3]
```diff
@@ -3,6 +3,10 @@ module.exports = {
   preset: 'ts-jest',
   testEnvironment: 'node',
   testTimeout: 30000,
+  transform: {
+    '^.+\\.ts?$': 'ts-jest',
+    "^.+\\.(js|jsx)$": "babel-jest"
```
Comment (dexturr) [inline jest.config-e2e.js:8]: question (blocking): why is this needed? I'd assume that ts-jest could handle all js files and we have no jsx files, only tsx?

Comment (dalebennett1992) [reply jest.config-e2e.js:8]: ah this is left over from something old I was trying. My bad.

File: test/e2e/wallet-helpers/api-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      return recoveryPhrase
+    })
+  }
+
+  async importWallet(recoveryPhrase: string) {
+    return (
+      await this.driver.executeScript<string>(async (recoveryPhrase: string) => {
+        // @ts-ignore
+        const resp = await window.client.request('admin.import_wallet', {
```
Comment (dexturr) [inline test/e2e/wallet-helpers/api-helpers.ts:21]: thought (non-blocking): having magic strings for method names is a bit suboptimal. Unfortunately because of scope/import issues we experienced you cannot import a map or anything here. I think out of scope for this PR but would like to see these shared between tests and FE code eventually.

Comment (dalebennett1992) [reply test/e2e/wallet-helpers/api-helpers.ts:19]: Yup, love this

File: package.json (hunk 1 above)
Comment (dalebennett1992) [reply package.json:64]: welp, it hated my yarn.lock! Moaning about strip-ansi in node modules. Do you guys use that? It is one of the changes in the yarn.lock. It builds locally absolutely fine for me even after nuking node modules and running from fresh.

File: test/e2e/wallet-helpers/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import JSONRPCClient from '../../../src/lib/json-rpc-client'
```
Comment (dexturr) [inline test/e2e/wallet-helpers/index.d.ts:1]: chore (non-blocking): can this be moved to src/index.d.ts as this now holds true for both the tests and the frontend build so would be more correct there

**Review by dexturr — APPROVED** [review]

File: src/contexts/json-rpc/json-rpc-provider.tsx [hunk 6]
```diff
@@ -15,6 +15,8 @@ const createClient = (logger: ReturnType<typeof useLogger>) => {
       backgroundPort.postMessage(msg)
     }
   })
+  // @ts-ignore
```
Comment (dexturr) [inline src/contexts/json-rpc/json-rpc-provider.tsx:18]: If we do this: https://github.com/vegaprotocol/vegawallet-browser/pull/208/files#r1196364639

we can remove this

```suggestion
```
